jetty-documentation/src/main/asciidoc/development/handlers/writing-custom-handlers.adoc

```diff
@@ -170,6 +170,27 @@ This is typically used to pass a request to a link:{JDURL}/org/eclipse/jetty/ser
 * link:{JDURL}/org/eclipse/jetty/server/handler/HandlerList.html[`HandlerList`] - A list of handlers that are called in turn until the request state is set as handled.
 * link:{JDURL}/org/eclipse/jetty/server/handler/ContextHandlerCollection.html[`ContextHandlerCollection`] - A collection of Handlers, of which one is selected by best match for the context path.
 
+[[injecting-handlers]]
+==== Injecting Handlers
+
+A custom handler can be injected using an injection based XML:
+[source, xml]
+----
+<?xml version="1.0"?>
+<!DOCTYPE Configure PUBLIC "-//Jetty//Configure//EN" "http://www.eclipse.org/jetty/configure_9_3.dtd">
+<Configure id="Server" class="org.eclipse.jetty.server.Server">
+    <Call name="setHandler">
+        <Arg>
+            <New id="myCustomJettyHandler" class="com.my.handler.CustomJettyHandler" />
+        </Arg>
+    </Call>
+</Configure>
+----
+
+* The custom handler JAR should be available at `/path/to/myjettybase/lib/ext/`
+* The custom handler XML should be available at `/path/to/myjettybase/etc/`
+* The custom handler XML should be loaded into the jetty instance at the right point in the XML load order by declaring it to be used in a custom INI
```

Comment:
I think this paragraph might be too specific. I think the XML code example is good to have, but the description could be more general by referring to the existing relevant documentation.

The `Handler` needs to be added to the server classpath. https://www.eclipse.org/jetty/documentation/current/startup-classpath.html

Then it can be added to the server, either by overriding some existing XML configuration files such as `jetty.xml`, or by defining a custom module as described at https://www.eclipse.org/jetty/documentation/current/custom-modules.html.

Reply:
@lachlan-robert Thanks for your suggestion. Does this look good? If yes, I will commit this change.

==== Injecting Handlers

The `Handler` needs to be added to the server classpath as described [here](https://www.eclipse.org/jetty/documentation/current/startup-classpath.html).
Then it can be added to the server, either by overriding some existing XML configuration files such as `jetty.xml` as shown below, or by defining a custom module as described [here](https://www.eclipse.org/jetty/documentation/current/custom-modules.html).

```
XML Code here.
```

Reply:
Yep, the content looks good to me.

But maybe just wait to hear what @WalkerWatch has to say before making the changes. 
I think there is also some special way to have internal links to different sections of the documentation, instead of having the URL links.

Reply:
In `writing-custom-handlers.adoc` there is an example which should be similar:
```
See the section on xref:basic-architecture[] to understand more about Handlers vs. Servlets.
```

So maybe `xref:startup-classpath[]` and `xref:custom-modules[]` will do the links you need.

Reply:
I think `xref` is used to link to various sections within the same page.
For example, in the `writing-custom-handlers.doc` [page](https://github.com/eclipse/jetty.project/blob/jetty-9.4.x/jetty-documentation/src/main/asciidoc/development/handlers/writing-custom-handlers.adoc#writing-custom-handlers):
* Clicking on `[basic-architecture]` does NOT work.
* But clicking on `Generating a Response` in the `Handling Requests` section does work.

Reply:
I think that is just broken when you view it on GitHub, it seems to work on the documentation pages here https://www.eclipse.org/jetty/documentation/current/jetty-handlers.html

And I just checked it also works if you build locally and go to `/jetty-documentation/target/docbkx/html/jetty-handlers.html`. We want to avoid linking to the URL if possible because it will break if there are different versions of the documentation, say for jetty 9.4.x and 10.0.x.

Reply:
Makes sense. 👍  
@lachlan-roberts I have committed the changes.

Reply:
@lachlan-roberts Do the changes look good? Thanks.

Reply:
@calvincodes yes, sorry for the delay, I've been looking how this would fit in with the docs for jetty-10. So after I figure that out I will merge the PR.

Reply:
Sure. No worries. Let me know if that would require any changes to this PR and I can try to incorporate them. Thanks again. :)